### PR TITLE
Feature/io 720 analytics clashlytics

### PR DESCRIPTION
**Pull request by ekaterina-demeshko-idf**

### CODE REVIEW

**Review by nikolay-ahramovich-idf — APPROVED** [review]

File: packages/domain/lib/usecase/analytics_screen_usecase.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import 'package:domain/services/analytics_service.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class LogAnalyticsScreenUseCase implements UseCaseParams<String, Future<void>> {
```
Comment (andrei-lastochkin-idf) [inline packages/domain/lib/usecase/analytics_screen_usecase.dart:4]: what is difference between LogAnalyticsEventUseCase and LogAnalyticsScreenUseCase?

File: packages/presentation/lib/screen/details/details_bloc.dart [hunk 2]
```diff
@@ -10,8 +11,12 @@ import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 abstract class DetailsBloc extends Bloc<DetailsScreenArguments, DetailsData> {
   factory DetailsBloc(
     GetCastUseCase getCastUseCase,
+    LogAnalyticsEventUseCase logAnalyticsEventUseCase,
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screen/details/details_bloc.dart]: move to base bloc
no need to inject it in each screen as u'll use it everywhere

File: packages/presentation/lib/screen/details/details_bloc.dart [hunk 3]
```diff
@@ -36,6 +45,7 @@ class _DetailsBloc extends BlocImpl<DetailsScreenArguments, DetailsData>
 
   @override
   Future<void> onShareButtonPressed(String message) async {
+    _logAnalyticsEventUseCase('share_button');
```
Comment (andrei-lastochkin-idf) [inline packages/presentation/lib/screen/details/details_bloc.dart]: create enum AnalyticsEventType and store events here or at least move event name to const

**Review by andrei-lastochkin-idf — APPROVED** [review]

File: packages/presentation/lib/screen/details/details_bloc.dart (hunk 2 above)
Comment (ekaterina-demeshko-idf) [reply packages/presentation/lib/screen/details/details_bloc.dart]: Not really everywhere. For example there is no events on Splash screen. Should it be in the base bloc anyway?

File: packages/domain/lib/usecase/analytics_screen_usecase.dart (hunk 1 above)
Comment (ekaterina-demeshko-idf) [reply packages/domain/lib/usecase/analytics_screen_usecase.dart:4]: LogAnalyticsEventUseCase send just event name, LogAnalyticsScreenUseCase send parameter of screen name using one event - screen_view

File: packages/presentation/lib/screen/details/details_bloc.dart (hunk 2 above)
Comment (andrei-lastochkin-idf) [reply packages/presentation/lib/screen/details/details_bloc.dart]: I think SplashScreen no worse than any other. Don't u log any event here?
In practice analytics exists everywhere. Even if 9/10 screens(except splash) use analytics, then we move it in base bloc

File: lib/main.dart [hunk 4]
```diff
@@ -11,6 +13,12 @@ import 'di/app_injector.dart';
 void mainCommon({Environment env = Environment.prod}) async {
   WidgetsFlutterBinding.ensureInitialized();
   await Firebase.initializeApp();
+  if (!kDebugMode) {
+    FlutterError.onError = FirebaseCrashlytics.instance.recordFlutterError;
+  }
+  if(kDebugMode) {
+    await FirebaseCrashlytics.instance.setCrashlyticsCollectionEnabled(false);
+  }
```
Comment (sergei-mikhailovskii-idf) [inline lib/main.dart]:
```suggestion
FirebaseCrashlytics.instance.setCrashlyticsCollectionEnabled(kDebugMode);
```

I think it's enough

File: packages/domain/lib/usecase/analytics_event_usecase.dart [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import 'package:domain/services/analytics_service.dart';
+import 'package:domain/usecase/usecase.dart';
+
+class LogAnalyticsEventUseCase
```
Comment (sergei-mikhailovskii-idf) [inline packages/domain/lib/usecase/analytics_event_usecase.dart]: good job!

**Review by sergei-mikhailovskii-idf — APPROVED** [review]

**Pull request merged**